src/main/java/com/skhu/luxuryshop/user/service/UserSignupService.java

```diff
@@ -0,0 +1,38 @@
+package com.skhu.luxuryshop.user.service;
+
+import com.skhu.luxuryshop.user.entity.UserEntity;
+import com.skhu.luxuryshop.user.dto.UserSignupDto;
+import com.skhu.luxuryshop.user.exception.DuplicatedEmailException;
+import com.skhu.luxuryshop.user.exception.FailedToSignupException;
+import com.skhu.luxuryshop.user.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class UserSignupService {
+    private final UserRepository userRepository;
+
+    //회원생성
+    @Transactional
+    public void save(UserSignupDto userSignupDto) {
+        UserEntity user = userSignupDto.toUserEntity();
+        if (!isDuplicatedEmail(userSignupDto.getEmail())) {
+            userRepository.save(user);
+            return;
+        }
+        throw new FailedToSignupException("회원가입 실패");
+    }
+
+    //Email 중복검사
+    public boolean isDuplicatedEmail(String email) {
+        Optional<UserEntity> user = userRepository.findByEmail(email);
+        if (user.isPresent()) {
+            throw new DuplicatedEmailException("중복된 이메일");
+        }
+        return false;
+    }
+}
```

Comment:
중복된 이메일이라면 exception이 두 번 throw 하게 되있네요! 실제로는 DuplicatedEmailException만 터질거구요!
위 리뷰에도 말씀드렸듯이 검증을 하는 메서드라면 boolean타입으로 메서드를 작성할  필요가 없습니다!
void로 하고 네이밍도 더 명확하게 할 수 있구요! 
그리고 Eamil 중복여부를 검사할거라면 JPA exist를 사용하는게 더 좋아보여요!
repository에서 findByEmail처럼 existByEmail로 간단하게 만들 수 있습니다.
아래처럼 수정할 수 있을 것 같아요!
```suggestion
    @Transactional
    public UserEntity save(UserSignupDto userSignupDto) {
        validateDuplicatedEmail(userSignupDto.getEmail());
        UserEntity user = userSignupDto.toUserEntity();
        return userRepository.save(user);
    }

    public void validateDuplicatedEmail(String email) {
        if(userRepository.existsByEamil(email)) {
            throw new DuplicatedEmailException("중복된 이메일");
        }
    }
```

Reply:
리뷰 확인했습니다. 수정 후 다시 커밋해볼게요!